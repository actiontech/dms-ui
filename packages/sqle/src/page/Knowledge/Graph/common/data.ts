import { SigmaContainerProps } from '@react-sigma/core';
import { EdgeType, NodeType } from '../index.type';
import { Attributes } from 'graphology-types';

export const sigmaSettings: SigmaContainerProps<
  NodeType,
  EdgeType,
  Attributes
>['settings'] = {
  // 允许在无效容器中渲染 (默认值: false)
  allowInvalidContainer: true,
  defaultNodeType: 'circle',
  // 设置边的默认类型 (默认值: 'line')
  defaultEdgeType: 'arrow',

  // 相机缩放比例限制 (默认值: min: 0.1, max: 32)
  minCameraRatio: 0.2,
  maxCameraRatio: 5,

  // 节点渲染程序类 (默认值: {})
  nodeProgramClasses: {},

  // 渲染性能优化
  hideEdgesOnMove: true, // 移动时隐藏边 (默认值: false)
  hideLabelsOnMove: false, // 移动时隐藏标签 (默认值: false)
  renderLabels: true, // 是否渲染标签 (默认值: true)
  renderEdgeLabels: false, // 是否渲染边标签 (默认值: false)

  // 标签渲染优化
  labelDensity: 0.1, // 标签密度 (默认值: 0.03)
  labelGridCellSize: 100, // 标签网格大小 (默认值: 100)
  labelRenderedSizeThreshold: 8, // 标签大小阈值 (默认值: 6)

  // 默认样式
  defaultNodeColor: '#1a73e8', // 节点颜色 (默认值: '#999')
  defaultEdgeColor: '#ccc', // 边颜色 (默认值: '#ccc')

  // 交互体验优化
  zoomToSizeRatioFunction: (x: number) => x // 缩放比例函数 (默认值: Math.pow)
};
