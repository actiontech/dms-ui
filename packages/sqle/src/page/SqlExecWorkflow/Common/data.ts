export const SOURCE_WORKFLOW_PATH_KEY = 'sourceWorkflowId';
