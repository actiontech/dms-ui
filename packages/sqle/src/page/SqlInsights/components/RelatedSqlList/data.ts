import { GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum } from '@actiontech/shared/lib/api/sqle/service/SqlInsight/index.enum';
import { t } from '../../../../locale';

export const RelatedSqlFilterSourceOptions = [
  {
    label: t('sqlInsights.relatedSqlList.source.order'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.workflow
  },
  {
    label: t('sqlInsights.relatedSqlList.source.sqlManage'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.sql_manage
  },
  {
    label: t('sqlInsights.relatedSqlList.source.workbench'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.workbench
  }
];
