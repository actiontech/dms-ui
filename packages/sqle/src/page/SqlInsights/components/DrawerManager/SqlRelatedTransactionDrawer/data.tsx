import { TransactionInfoLockTypeEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';
import { t } from '../../../../../locale';

export const TransactionInfoLockTypeEnumDict = {
  [TransactionInfoLockTypeEnum.EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.exclusive'
  ),
  [TransactionInfoLockTypeEnum.SHARED]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.shared'
  ),
  [TransactionInfoLockTypeEnum.INTENTION_SHARED]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.intentionShared'
  ),
  [TransactionInfoLockTypeEnum.INTENTION_EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.intentionExclusive'
  ),
  [TransactionInfoLockTypeEnum.SHARED_INTENTION_EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.sharedIntentionExclusive'
  ),
  [TransactionInfoLockTypeEnum.ROW_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.rowLock'
  ),
  [TransactionInfoLockTypeEnum.TABLE_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.tableLock'
  ),
  [TransactionInfoLockTypeEnum.METADATA_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.metadataLock'
  )
};

export const TransactionInfoLockTypeEnumFlagDict = {
  [TransactionInfoLockTypeEnum.SHARED]: 'S',
  [TransactionInfoLockTypeEnum.INTENTION_SHARED]: 'IS',
  [TransactionInfoLockTypeEnum.INTENTION_EXCLUSIVE]: 'IX',
  [TransactionInfoLockTypeEnum.SHARED_INTENTION_EXCLUSIVE]: 'SIX',
  [TransactionInfoLockTypeEnum.ROW_LOCK]: 'RL',
  [TransactionInfoLockTypeEnum.TABLE_LOCK]: 'TL',
  [TransactionInfoLockTypeEnum.METADATA_LOCK]: 'ML',
  [TransactionInfoLockTypeEnum.EXCLUSIVE]: 'X'
};
