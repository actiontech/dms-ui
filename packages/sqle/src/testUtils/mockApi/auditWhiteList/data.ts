import { IAuditWhitelistResV1 } from '@actiontech/shared/lib/api/sqle/service/common';
import { CreateAuditWhitelistReqV1MatchTypeEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';

export const auditWhiteListMockData: IAuditWhitelistResV1[] = [
  {
    audit_whitelist_id: 1,
    desc: 'test1',
    match_type: CreateAuditWhitelistReqV1MatchTypeEnum.exact_match,
    value: 'SELECT 1;',
    matched_count: 1,
    last_match_time: '2024-08-22T11:04:29.543+08:00'
  },
  {
    audit_whitelist_id: 2,
    desc: 'test2',
    match_type: CreateAuditWhitelistReqV1MatchTypeEnum.fp_match,
    value: 'SELECT 1;',
    matched_count: 10,
    last_match_time: undefined
  },
  {
    audit_whitelist_id: 3,
    desc: 'test3',
    match_type: CreateAuditWhitelistReqV1MatchTypeEnum.exact_match,
    value: 'SELECT 1;',
    matched_count: 0
  },
  {
    audit_whitelist_id: 4,
    desc: 'test4'
  }
];
