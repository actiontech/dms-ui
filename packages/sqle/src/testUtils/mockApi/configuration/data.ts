/**
 *  Generated by actionsky-cli api-mocks, do not modify this file unless necessary.
 *  When modifying the file, please also update the corresponding unit test files.
 */

import { ISSHPublicKeyInfo } from '@actiontech/shared/lib/api/sqle/service/common.d';

export const deriversMockData = [
  {
    driver_name: 'mysql',
    default_port: 3000,
    logo_url: 'test/logo?instance_type=mysql'
  }
];

export const testGitConnectionMockData = {
  is_connected_success: true,
  error_message: null,
  branches: ['main', 'test']
};

export const getSSHPublicKeyMockData = {
  public_key:
    'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDzBOtcopZ1IdU51augExmKzRRYsJdNDnxVf5mlCRch2hMq4y02F4q4T71WFoLg22FptnxmY+LkOPgYyIvl6MdfBS6RQarc3aC2rO5bCbzuMf2f6+2hbYH3ACXJWAn4xX8b0ZK25t1u8jrXmAJPVXIMLEcLL4Wo7IqCQ1PwrkIeXuMfznHzVoAEv5o+jbUkj++I46AEmqrmDsgWCt83JK7ctBzhEXJWhF5rSyqGxXBptVPVMEO4ShJi2rKdk+ZbBGIAGu0db72hlvdqzUeU2Ng+J0dcIDsS/4pmEtQ+6uJFgslqF2Vt708tcOPTG4qdfFaK453PyceZ/F/f/BN+bPDJCh+oPQ3zdsKHmh0rXefICPMWe73QRLAyW/JeWcsnFlgoBmlwB4++Isc6xYx0dHFM/ws10vTJDif3CAi0VE1vKdSZvVxTFq5bZKhTX3zfA24yZzAl9BCOrQzWh20NWJ3Lgwk2JWL3NVsChKNk4aTCHNz1OyW/nojrd5tCyyqF06OqG1/xc5RmRkCH9gXqFA0Ngf3IdxJdpL9q3G60CBAi7Gfs6+gLIlMc6sgnlZdnyCj3Gmh7yRrmMZUDAmZnyvrYw3+4qQcVBi5gw64wuya2K6Hz6lrgxT9AlcUTNdHn1W6TimZtO3JUDclMUbA6euN/NOFsY4VpYPVNoMND3Lwesw=='
} as ISSHPublicKeyInfo;
