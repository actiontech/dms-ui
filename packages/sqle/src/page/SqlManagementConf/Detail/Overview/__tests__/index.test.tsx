import { mockUseCurrentProject } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentProject';
import ConfDetailOverview from '..';
import { superRender } from '../../../../../testUtils/customRender';
import { SQL_MANAGEMENT_CONF_OVERVIEW_TAB_KEY } from '../../index.data';
import { mockUseCurrentUser } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentUser';
import instanceAuditPlan from '../../../../../testUtils/mockApi/instanceAuditPlan';
import { act, fireEvent } from '@testing-library/react';
import { mockProjectInfo } from '@actiontech/shared/lib/testUtil/mockHook/data';
import { mockInstanceAuditPlanInfo } from '../../../../../testUtils/mockApi/instanceAuditPlan/data';
import {
  InstanceAuditPlanInfoActiveStatusEnum,
  UpdateAuditPlanStatusReqV1ActiveEnum
} from '@actiontech/shared/lib/api/sqle/service/common.enum';

describe('test Overview', () => {
  const handleChangeTabSpy = jest.fn();
  const instanceAuditPlanId = '1';
  const refreshAuditPlanDetailSpy = jest.fn();
  const customRender = (hasOpPermission = true) => {
    return superRender(
      <ConfDetailOverview
        activeTabKey={SQL_MANAGEMENT_CONF_OVERVIEW_TAB_KEY}
        handleChangeTab={handleChangeTabSpy}
        instanceAuditPlanId={instanceAuditPlanId}
        refreshAuditPlanDetail={refreshAuditPlanDetailSpy}
        hasOpPermission={hasOpPermission}
      />
    );
  };
  let getInstanceAuditPlanOverviewSpy: jest.SpyInstance;

  beforeEach(() => {
    jest.useFakeTimers();
    mockUseCurrentProject();
    mockUseCurrentUser();
    getInstanceAuditPlanOverviewSpy =
      instanceAuditPlan.getInstanceAuditPlanOverview();
  });
  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });
  it('should request data for the table when the component is active', async () => {
    customRender();

    await act(async () => jest.advanceTimersByTime(3000));

    expect(getInstanceAuditPlanOverviewSpy).toHaveBeenCalledTimes(1);
    expect(getInstanceAuditPlanOverviewSpy).toHaveBeenNthCalledWith(1, {
      project_name: mockProjectInfo.projectName,
      instance_audit_plan_id: instanceAuditPlanId
    });
  });

  it('should enable, disable, and delete actions with correct API calls and refresh the table', async () => {
    const updateAuditPlanStatusSpy = instanceAuditPlan.updateAuditPlanStatus();
    const deleteAuditPlanByTypeSpy = instanceAuditPlan.deleteAuditPlanByType();
    const { getAllByText, findByText, getByText } = customRender();

    await act(async () => jest.advanceTimersByTime(3000));

    expect(getAllByText('停 用').length).toBe(
      mockInstanceAuditPlanInfo.filter(
        (v) => v.active_status === InstanceAuditPlanInfoActiveStatusEnum.normal
      ).length
    );
    expect(getAllByText('启 用').length).toBe(
      mockInstanceAuditPlanInfo.filter(
        (v) =>
          v.active_status === InstanceAuditPlanInfoActiveStatusEnum.disabled
      ).length
    );
    expect(getAllByText('删 除').length).toBe(mockInstanceAuditPlanInfo.length);

    // disabled
    fireEvent.click(getAllByText('停 用')[0]);
    await findByText(
      '停用后，将不再扫描该类型的SQL，当前数据会被保留，是否确认停用？'
    );
    fireEvent.click(getByText('确 认'));

    expect(updateAuditPlanStatusSpy).toHaveBeenCalledTimes(1);
    expect(updateAuditPlanStatusSpy).toHaveBeenNthCalledWith(1, {
      project_name: mockProjectInfo.projectName,
      instance_audit_plan_id: instanceAuditPlanId,
      audit_plan_id: mockInstanceAuditPlanInfo
        .find(
          (v) =>
            v.active_status === InstanceAuditPlanInfoActiveStatusEnum.normal
        )
        ?.audit_plan_type?.audit_plan_id?.toString(),
      active: UpdateAuditPlanStatusReqV1ActiveEnum.disabled
    });
    expect(getAllByText('停 用')[0].closest('button')).toBeDisabled();

    await act(async () => jest.advanceTimersByTime(3000));
    expect(getAllByText('停 用')[0].closest('button')).not.toBeDisabled();
    expect(getByText('停用成功！')).toBeInTheDocument();
    expect(getInstanceAuditPlanOverviewSpy).toHaveBeenCalledTimes(2);

    // enabled
    fireEvent.click(getAllByText('启 用')[0]);
    expect(updateAuditPlanStatusSpy).toHaveBeenCalledTimes(2);
    expect(updateAuditPlanStatusSpy).toHaveBeenNthCalledWith(2, {
      project_name: mockProjectInfo.projectName,
      instance_audit_plan_id: instanceAuditPlanId,
      audit_plan_id: mockInstanceAuditPlanInfo
        .find(
          (v) =>
            v.active_status === InstanceAuditPlanInfoActiveStatusEnum.disabled
        )
        ?.audit_plan_type?.audit_plan_id?.toString(),
      active: UpdateAuditPlanStatusReqV1ActiveEnum.normal
    });

    expect(getAllByText('启 用')[0].closest('button')).toBeDisabled();
    await act(async () => jest.advanceTimersByTime(3000));
    expect(getAllByText('启 用')[0].closest('button')).not.toBeDisabled();
    expect(getByText('启用成功！')).toBeInTheDocument();
    expect(getInstanceAuditPlanOverviewSpy).toHaveBeenCalledTimes(3);

    //delete
    fireEvent.click(getAllByText('删 除')[0]);
    await findByText('删除后该类型智能扫描数据将不再被保留，是否确认删除？');
    fireEvent.click(getAllByText('确 认')[1]);
    expect(deleteAuditPlanByTypeSpy).toHaveBeenCalledTimes(1);
    expect(deleteAuditPlanByTypeSpy).toHaveBeenNthCalledWith(1, {
      project_name: mockProjectInfo.projectName,
      instance_audit_plan_id: instanceAuditPlanId,
      audit_plan_id:
        mockInstanceAuditPlanInfo[0].audit_plan_type?.audit_plan_id?.toString()
    });

    expect(getAllByText('删 除')[0].closest('button')).toBeDisabled();
    await act(async () => jest.advanceTimersByTime(3000));
    expect(getAllByText('删 除')[0].closest('button')).not.toBeDisabled();
    expect(getByText('删除成功！')).toBeInTheDocument();
    expect(getInstanceAuditPlanOverviewSpy).toHaveBeenCalledTimes(4);
    expect(refreshAuditPlanDetailSpy).toHaveBeenCalledTimes(1);
  });

  it('should handle row click events to change the active tab', async () => {
    const { getByText } = customRender();
    await act(async () => jest.advanceTimersByTime(3000));

    fireEvent.click(
      getByText(mockInstanceAuditPlanInfo[0].audit_plan_type?.desc!)
    );

    expect(handleChangeTabSpy).toHaveBeenCalledTimes(1);
    expect(handleChangeTabSpy).toHaveBeenNthCalledWith(
      1,
      mockInstanceAuditPlanInfo[0].audit_plan_type?.audit_plan_id?.toString()
    );
  });

  it('should match snapshot when hasOpPermission is equal false', () => {
    const { container } = customRender(false);
    expect(container).toMatchSnapshot();
  });
});
