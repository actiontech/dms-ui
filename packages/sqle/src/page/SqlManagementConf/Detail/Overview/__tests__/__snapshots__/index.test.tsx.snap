// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Overview render abnormal status audit plan 1`] = `
<div>
  <div
    class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table-wrapper actiontech-table-namespace table-row-cursor css-fxd6ka css-dev-only-do-not-override-bj9uhl"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-bj9uhl"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-fixed-column ant-table-scroll-horizontal ant-table-has-fix-right"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                  style="overflow-x: auto; overflow-y: hidden;"
                >
                  <table
                    style="min-width: 100%; table-layout: auto;"
                  >
                    <colgroup>
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col
                        style="width: 246px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          智能扫描类型
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          审核规则模板
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          任务状态
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          连接信息
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          采集到的SQL数
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          审核有问题的SQL数
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          最近一次采集时间
                        </th>
                        <th
                          class="ant-table-cell actiontech-table-actions-column ant-table-cell-fix-right ant-table-cell-fix-right-first"
                          scope="col"
                          style="position: sticky; right: 0px;"
                        >
                          操作
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        aria-hidden="true"
                        class="ant-table-measure-row"
                        style="height: 0px; font-size: 0px;"
                      >
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <a
                            class="ant-typography css-dev-only-do-not-override-bj9uhl"
                          >
                            库表元数据
                          </a>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <a
                            href="/sqle/rule-manager/global-detail/default_MySQL/MySQL"
                            target="_blank"
                          >
                            default_MySQL
                          </a>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="css-10kos8e"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-bj9uhl ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <svg
                                  height="16"
                                  viewBox="0 0 14 14"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M7 12.833A5.833 5.833 0 1 1 7 1.167a5.833 5.833 0 0 1 0 11.666m0-1.166a4.666 4.666 0 1 0 0-9.333 4.666 4.666 0 0 0 0 9.333m-.583-7.584h1.166V5.25H6.417zm0 2.334h1.166v3.5H6.417z"
                                    fill="#EDB054"
                                  />
                                </svg>
                              </div>
                            </div>
                            <span>
                              运行异常
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          -
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          0
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          0
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          2024-08-06 11:30:43
                        </td>
                        <td
                          class="ant-table-cell actiontech-table-actions-column ant-table-cell-fix-right ant-table-cell-fix-right-first"
                          style="position: sticky; right: 0px;"
                        >
                          <div
                            class="css-new95e"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-bj9uhl ant-space-horizontal ant-space-align-center"
                            >
                              <div
                                class="ant-space-item"
                                style="margin-right: 8px;"
                              >
                                <button
                                  class="ant-btn css-dev-only-do-not-override-9n8g17 ant-btn-default ant-btn-sm actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                  disabled=""
                                  type="button"
                                >
                                  <span>
                                    启 用
                                  </span>
                                </button>
                              </div>
                              <div
                                class="ant-space-item"
                                style="margin-right: 8px;"
                              >
                                <button
                                  class="ant-btn css-dev-only-do-not-override-9n8g17 ant-btn-default ant-btn-sm actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                  type="button"
                                >
                                  <span>
                                    停 用
                                  </span>
                                </button>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <button
                                  class="ant-btn css-dev-only-do-not-override-9n8g17 ant-btn-default ant-btn-sm ant-btn-dangerous actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                  type="button"
                                >
                                  <span>
                                    删 除
                                  </span>
                                </button>
                              </div>
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <a
                            class="ant-typography css-dev-only-do-not-override-bj9uhl"
                          >
                            慢日志
                          </a>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <a
                            href="/sqle/project/700300/rule/template/detail/mysql_rule_template/MySQL"
                            target="_blank"
                          >
                            mysql_rule_template
                          </a>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="css-10kos8e"
                          >
                            <svg
                              fill="none"
                              height="16"
                              viewBox="0 0 16 16"
                              width="16"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M6.875 1.516a2.25 2.25 0 0 1 2.25 0l3.928 2.267a2.25 2.25 0 0 1 1.125 1.95v4.535a2.25 2.25 0 0 1-1.125 1.948l-3.928 2.269a2.25 2.25 0 0 1-2.25 0l-3.928-2.268a2.25 2.25 0 0 1-1.125-1.95V5.733a2.25 2.25 0 0 1 1.125-1.949zM8 10v2M8 4l.004 4.5"
                                stroke="#F59957"
                                stroke-width="1.5"
                              />
                            </svg>
                            <span>
                              未知状态...
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="css-1sg2lsz"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-bj9uhl ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  class="rect-border-wrapper css-1iiq8r2"
                                >
                                  <svg
                                    height="16"
                                    viewBox="0 0 16 16"
                                    width="16"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M11.333 9.333H8.44A4.002 4.002 0 0 1 .667 8a4 4 0 0 1 7.772-1.333h6.894v2.666H14V12h-2.667zm-6.666 0a1.333 1.333 0 1 0 0-2.666 1.333 1.333 0 0 0 0 2.666"
                                      fill="#8A8F99"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                            <span
                              class="ant-typography css-dev-only-do-not-override-bj9uhl"
                            >
                              <div
                                aria-label="Copy"
                                class="ant-typography-copy"
                                role="button"
                                style="border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;"
                                tabindex="0"
                              >
                                <span
                                  aria-label="copy"
                                  class="anticon anticon-copy"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="copy"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          0
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          0
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          -
                        </td>
                        <td
                          class="ant-table-cell actiontech-table-actions-column ant-table-cell-fix-right ant-table-cell-fix-right-first"
                          style="position: sticky; right: 0px;"
                        >
                          <div
                            class="css-new95e"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-bj9uhl ant-space-horizontal ant-space-align-center"
                            >
                              <div
                                class="ant-space-item"
                                style="margin-right: 8px;"
                              >
                                <button
                                  class="ant-btn css-dev-only-do-not-override-9n8g17 ant-btn-default ant-btn-sm actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                  disabled=""
                                  type="button"
                                >
                                  <span>
                                    启 用
                                  </span>
                                </button>
                              </div>
                              <div
                                class="ant-space-item"
                                style="margin-right: 8px;"
                              >
                                <button
                                  class="ant-btn css-dev-only-do-not-override-9n8g17 ant-btn-default ant-btn-sm actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                  disabled=""
                                  type="button"
                                >
                                  <span>
                                    停 用
                                  </span>
                                </button>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <button
                                  class="ant-btn css-dev-only-do-not-override-9n8g17 ant-btn-default ant-btn-sm ant-btn-dangerous actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                  type="button"
                                >
                                  <span>
                                    删 除
                                  </span>
                                </button>
                              </div>
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test Overview should match snapshot when hasOpPermission is equal false 1`] = `
<div>
  <div
    class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table-wrapper actiontech-table-namespace table-row-cursor css-fxd6ka css-dev-only-do-not-override-bj9uhl"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-bj9uhl"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-empty ant-table-fixed-column ant-table-scroll-horizontal ant-table-has-fix-right"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                  style="overflow-x: auto; overflow-y: hidden;"
                >
                  <table
                    style="min-width: 100%; table-layout: auto;"
                  >
                    <colgroup>
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col
                        style="width: 246px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          智能扫描类型
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          审核规则模板
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          任务状态
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          连接信息
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          采集到的SQL数
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          审核有问题的SQL数
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          最近一次采集时间
                        </th>
                        <th
                          class="ant-table-cell actiontech-table-actions-column ant-table-cell-fix-right ant-table-cell-fix-right-first"
                          scope="col"
                          style="position: sticky; right: 0px;"
                        >
                          操作
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        aria-hidden="true"
                        class="ant-table-measure-row"
                        style="height: 0px; font-size: 0px;"
                      >
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-placeholder"
                      >
                        <td
                          class="ant-table-cell"
                          colspan="8"
                        >
                          <div
                            class="ant-table-expanded-row-fixed"
                            style="width: 0px; position: sticky; left: 0px; overflow: hidden;"
                          />
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
